lib/connection.cpp

```diff
@@ -317,6 +317,11 @@ class Connection::Private {
             };
             auto job = q->callApi<ClaimKeysJob>(hash);
             connect(job, &BaseJob::finished, q, [this, deviceId, job, senderId] {
+                static QSet<std::pair<QString, QString>> triedDevices;
```

Comment:
`static` is not very good here. Imagine two different accounts participating in the same room: if one of them tries a given pair another one won't ever have a chance to request his copy of Olm sessions - unless I miss something.

Reply:
Good point; fixed